app/queries/document/paginated_timeline_query.rb

```diff
@@ -0,0 +1,111 @@
+class Document
```

Comment:
I'm still finding this file a bit tricky to parse, and I think it's because this query object actually represents four(?) different queries which don't actually need to depend on each other. Would it make sense to lift the remarks and versions queries up to the Document model and pass in the remark/version ID lists as a parameter? Otherwise those queries are quite simple. Then we could leave this file to focus on the gnarly pagination/union behaviour.